pkg/resizer/csi_resizer_test.go

```diff
@@ -66,8 +71,218 @@ func TestNewResizer(t *testing.T) {
+			nodeResizeRequired: true,
+			err:                errors.New("volume testEBSPV is not migrated to CSI"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			driverName := tc.driverName
+			client := csi.NewMockClient(driverName, true, true, true)
+			k8sClient, informerFactory := fakeK8s()
+			resizer, err := NewResizerFromClient(client, 0, k8sClient, informerFactory)
+			if err != nil {
+				t.Fatalf("Failed to create resizer: %v", err)
+			}
+
+			pv := tc.pv
+			expectedSize := quantityGB(2)
+			newSize, nodeResizeRequired, err := resizer.Resize(pv, expectedSize)
+
+			if tc.err != nil {
+				if err == nil {
+					t.Fatalf("Got wrong error, wanted: %v, got: %v", tc.err, err)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("Failed to resize the PV: %v", err)
+				}
+
+				if newSize != expectedSize {
+					t.Fatalf("newSize mismatches, wanted: %v, got: %v", expectedSize, newSize)
+				}
+				if nodeResizeRequired != tc.nodeResizeRequired {
+					t.Fatalf("nodeResizeRequired mismatches, wanted: %v, got: %v", tc.nodeResizeRequired, nodeResizeRequired)
+				}
+			}
+		})
+	}
+}
+
+func TestCanSupport(t *testing.T) {
+	testCases := []struct {
+		name       string
+		driverName string
+		pv         *v1.PersistentVolume
+		pvc        *v1.PersistentVolumeClaim
+		canSupport bool
+	}{
+		{
+			name:       "EBS PV/PVC is supported",
+			driverName: "ebs.csi.aws.com",
+			pv:         createEBSPV(1),
+			pvc:        createPVC(2, 1, "ebs.csi.aws.com"),
+			canSupport: true,
+		},
+		{
+			name:       "EBS PV/PVC is not supported when migartion is disabled",
+			driverName: "ebs.csi.aws.com",
+			pv:         createEBSPV(1),
+			pvc:        createPVC(2, 1, "kubernetes.io/aws-ebs"),
+			canSupport: false,
+		},
+		{
+			name:       "PD PV/PVC is supported",
+			driverName: "pd.csi.storage.gke.io",
+			pv:         createGCEPDPV(1),
+			pvc:        createPVC(2, 1, "pd.csi.storage.gke.io"),
+			canSupport: true,
+		},
+		{
+			name:       "unknown PV/PVC is not supported",
+			driverName: "ebs.csi.aws.com",
+			pv:         createEBSPV(1),
+			pvc:        createPVC(2, 1, "unknown"),
+			canSupport: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			driverName := tc.driverName
+			client := csi.NewMockClient(driverName, true, true, true)
+			k8sClient, informerFactory := fakeK8s()
+			resizer, err := NewResizerFromClient(client, 0, k8sClient, informerFactory)
+			if err != nil {
+				t.Fatalf("Failed to create resizer: %v", err)
+			}
+
+			canSupport := resizer.CanSupport(tc.pv, tc.pvc)
+			if canSupport != tc.canSupport {
+				t.Fatalf("Wrong canSupport, wanted: %v got: %v", tc.canSupport, canSupport)
+			}
+		})
+	}
+}
+
+func quantityGB(i int) resource.Quantity {
+	q := resource.NewQuantity(int64(i*1024*1024), resource.BinarySI)
+	return *q
+}
+
+func createPVC(requestGB, capacityGB int, resizerName string) *v1.PersistentVolumeClaim {
```

Comment:
seems like `requestGB` and `capacityGB` are not necessary inputs to this function. We don't actually test different values for these so we could just make them consts. Ditto throughout for the capacity numbers we're passing in. Everything could just be one flat number.

If you *must* pass them in, lets use consts at the caller level so it's easy to tell what each number means